web/src/components/task/task-sidebar/finish-button/finish-button.tsx

```diff
@@ -52,18 +52,23 @@ export const FinishButton: FC<TaskSidebarProps> = ({
 
     const handleFinishValidation = isSplitValidation ? onFinishSplitValidation : onFinishValidation;
 
-    let isDisabled = !isAnnotatable;
-    if (jobType !== 'extensive_coverage') {
-        isDisabled =
-            !isAnnotatable ||
-            !allValidated ||
-            (isValidation && !touchedPagesCount && !editedPagesCount);
-    }
+    const isDisabled = useMemo(() => {
```

Comment:
Which reason is to use `useMemo` here? The logic is not resource-intensive here, we can just create a variable and save there needed condition

Reply:
True, addresed